### PR TITLE
Prevent docker-compose.yml and Dockerfile from being written

### PR DESCRIPTION
**Pull request by erik-megarad**

### CODE REVIEW

**Review by ntindle — DISMISSED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -10,4 +10,6 @@ services:
       - .env
     volumes:
       - ./:/app
+      - ./docker-compose.yml:/app/docker-compose.yml:ro
+      - ./Dockerfile:/app/Dockerfile.yml:ro
```
Comment (Pwuts) [inline docker-compose.yml]: Typo: `/app/Dockerfile.yml` ;)

To prevent docker from creating a folder when a file does not exist, use a bind mount:
```suggestion
      - type: bind
        source: ./docker-compose.yml
        target: /app/docker-compose.yml
        read_only: true
      - type: bind
        source: ./Dockerfile
        target: /app/Dockerfile
        read_only: true
```

I hope that this can be combined with the existing `./:/app` mount.

In principle we have no reason to mount these docker-related files at all, but the only alternative (afaik) is to explicitly specify all the folders and files that should be mounted, and that will probably be a pretty long list.

Comment (lc0rp) [reply docker-compose.yml]: The relative paths (./docker-compose.yml) should create bind mounts.

So, I believe these are effectively the same thing:

```
volumes:
    - ./docker-compose.yml:...
```

and 
```
volumes:
   - type: bind
      source: ./docker-compose.yml
      ...
 ```

Comment (Pwuts) [reply docker-compose.yml]: From the [docs](https://docs.docker.com/compose/compose-file/compose-file-v3/#:~:text=When%20creating%20bind%20mounts%2C%20using%20the%20long%20syntax%20requires%20the%20referenced%20folder%20to%20be%20created%20beforehand.%20Using%20the%20short%20syntax%20creates%20the%20folder%20on%20the%20fly%20if%20it%20doesn%E2%80%99t%20exist.%20See%20the%20bind%20mounts%20documentation%20for%20more%20information.):
> When creating bind mounts, using the long syntax requires the referenced folder to be created beforehand. Using the short syntax creates the folder on the fly if it doesn’t exist. See the [bind mounts documentation](https://docs.docker.com/storage/bind-mounts/#differences-between--v-and---mount-behavior) for more information.

TL;DR: no, they are not the same thing. At least they shouldn't be. Docker Desktop for Windows has deviant behavior, where the long form with `type: bind` behaves the same way as the short form. See https://github.com/docker/for-win/issues/11958

**Review by Pwuts — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (Pwuts) [reply docker-compose.yml]: Resolving this convo so we can merge
